### ISSUE TITLE
⌨ Multiselect combobox items set aria-selected on the highlighted item

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#8424]
## ⌨ Accessibility Issue

The manual accessibility check of the 'SkillBrowser' component (Issue [#8001](https://github.com/GCTC-NTGC/gc-digital-talent/issues/8001)) found that its Multiselect combobox items set aria-selected on the highlighted item, not on selected items. This is a bug with downshift.  We may need to hack around this. 

### Accessible Rich Internet Applications (WAI-ARIA) Criteria 

[6.5.1. Widget Attributes  - ARIA selected states](https://www.w3.org/TR/wai-aria-1.0/states_and_properties#aria-selected)

## 🦋 Expected Behaviour

Only selected items are tagged as an [aria-selected (state)](https://www.w3.org/TR/wai-aria-1.0/states_and_properties#aria-selected). 

## 🕵️ Details

Add any additional details that could assist with troubleshooting/fixing the issue.

- **Operating System**: Windows, MacOS.
- **Browser**: Chrome, Edge, Firefox and Safari.
- **Assistive Technology**: NVDA, Narrator and VoiceOver

## 📸 Screenshot

![Image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/4848a39f-4e43-47e2-99bb-e051cb57dcbf)

## 🙋‍♀️ Proposed Solution

Our multi select list should have a aria-multiselectable=true on the ul[role=listbox], items in the list should have aria-selected values set to true when selected and false when not selected.

This is likely a high priority since screen readers announce that the item is selected regardless of its actual state since downshift adds aria-selected=true to the highlighted item which is very confusing.

## ✅ Acceptance Criteria

- [ ] Only selected items have ARIA selected state set to true.
